flow/cmd/maintenance.go

```diff
@@ -0,0 +1,212 @@
+package cmd
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+
+	"go.temporal.io/sdk/client"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/PeerDB-io/peer-flow/generated/protos"
+	"github.com/PeerDB-io/peer-flow/peerdbenv"
+	"github.com/PeerDB-io/peer-flow/shared"
+	peerflow "github.com/PeerDB-io/peer-flow/workflows"
+)
+
+type MaintenanceCLIParams struct {
+	TemporalHostPort         string
+	TemporalNamespace        string
+	Mode                     string
+	OutputFile               string
+	InputFile                string
+	FlowGrpcAddress          string
+	FlowTlsEnabled           bool
+	SkipOnApiVersionMatch    bool
+	SkipOnNoMirrors          bool
+	UserMaintenanceTaskQueue bool
+}
+
+type StartMaintenanceResult struct {
+	APIVersion    string `json:"apiVersion,omitempty"`
+	CLIVersion    string `json:"cliVersion,omitempty"`
+	SkippedReason string `json:"skippedReason,omitempty"`
+	Skipped       bool   `json:"skipped,omitempty"`
+}
+
+// MaintenanceMain is the entry point for the maintenance command, requires access to Temporal client, will exit after
+// running the requested maintenance workflow
+func MaintenanceMain(ctx context.Context, args *MaintenanceCLIParams) error {
+	clientOptions := client.Options{
+		HostPort:  args.TemporalHostPort,
+		Namespace: args.TemporalNamespace,
+		Logger:    slog.New(shared.NewSlogHandler(slog.NewJSONHandler(os.Stdout, nil))),
+	}
+
+	tc, err := setupTemporalClient(ctx, clientOptions)
+	if err != nil {
+		return fmt.Errorf("unable to create Temporal client: %w", err)
+	}
+
+	taskQueueId := shared.MaintenanceFlowTaskQueue
+	if !args.UserMaintenanceTaskQueue {
+		taskQueueId = shared.PeerFlowTaskQueue
+	}
+
+	if args.Mode == "start" {
+		skipped, err := skipStartMaintenanceIfNeeded(ctx, args)
+		if err != nil {
+			return err
+		}
+		if skipped {
+			return nil
+		}
+		slog.Info("Running start maintenance workflow")
+		workflowRun, err := peerflow.RunStartMaintenanceWorkflow(ctx, tc, &protos.StartMaintenanceFlowInput{}, taskQueueId)
+		if err != nil {
+			slog.Error("Error running start maintenance workflow", "error", err)
+			return err
+		}
+		var output *protos.StartMaintenanceFlowOutput
+		err = workflowRun.Get(ctx, &output)
+		if err != nil {
+			slog.Error("Error in start maintenance workflow", "error", err)
+			return err
+		}
+		slog.Info("Start maintenance workflow completed", "output", output)
+		return WriteMaintenanceOutput(args.OutputFile, StartMaintenanceResult{
```

Comment:
any reason why you're not preferring catalog for this?

Reply:
Reason at that time was possibility of using configmap and local debugging, replaced with catalog